compiler/wasm/src/compile.rs

```diff
@@ -24,14 +28,26 @@ extern "C" {
     pub type StringArray;
 }
 
+#[derive(Deserialize)]
+struct DependencyGraph {
+    root_dependencies: Vec<String>,
+    library_dependencies: HashMap<String, Vec<String>>,
```

Comment:
```suggestion
    root_dependencies: Vec<CrateName>,
    library_dependencies: HashMap<CrateName, Vec<CrateName>>,
```

We want to enforce that these are value crate names.